### PR TITLE
Add a page to view screenshots from webdriver test

### PR DESCRIPTION
**Pull request by shifucun**
Page /screenshot/<folder> will load screenshot images from gs://datcom-staging-screenshot/<folder> and display them.

This will be used in webdriver test before PR commit and push. Author will receive an email with links of this page to verify the screenshots.

### CODE REVIEW

File: server/__init__.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def create_app():
     cache.init_app(app)
 
     # apply the blueprints to the app
-    from routes import browser, factcheck, redirects, placelist, static, tools
+    from routes import browser, factcheck, redirects, placelist, static, tools, dev
```
Comment (beets) [inline server/__init__.py:46]: Please keep alphabetical ordering here and in the register_blueprints section

File: server/lib/gcs.py [hunk 2]
```diff
@@ -14,34 +14,54 @@
...
+
+
+def list_png(bucket_name, prefix):
+    """Return a list of images in a given bucket and folder prefix.
+
+    Args:
+      bucket_name: the bucket where the image is stored.
+      prefix: the folder prefix
+    Returns:
+      An array of 64-encoded images.
```
Comment (beets) [inline server/lib/gcs.py:58]: base 64

File: server/templates/dev/screenshot.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#}
+{% extends 'base.html' %}
+
+{% set main_id = 'screenshot' %}
+{% set title = 'Screenshot page' %}
+
+{% block head %}
+  <link rel="stylesheet" href={{url_for('static', filename='css/dev.min.css')}}>
+  <style>
+  </style>
```
Comment (beets) [inline server/templates/dev/screenshot.html:24]: unnecessary style block

File: server/lib/gcs.py [hunk 4]
```diff
@@ -14,34 +14,54 @@
...
+    """
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(bucket_name)
+    blobs = bucket.list_blobs(prefix=prefix)
```
Comment (beets) [inline server/lib/gcs.py:62]: Any security concern from listing blobs based on user input?

File: server/__init__.py (hunk 1 above)
Comment (shifucun) [reply server/__init__.py]: Done

File: server/lib/gcs.py (hunk 2 above)
Comment (shifucun) [reply server/lib/gcs.py]: Done

File: server/lib/gcs.py (hunk 4 above)
Comment (shifucun) [reply server/lib/gcs.py:62]: The listing here restricts to this bucket only which only contains the screenshot images. It seems okay to accept the prefix from user then.

File: server/templates/dev/screenshot.html (hunk 3 above)
Comment (shifucun) [reply server/templates/dev/screenshot.html]: Done

**Comment by shifucun** [comment]
> How will we see whether any diffs were generated - should we show side-by-side images between runs? Will we have to fix post-commit with a new PR? Have we explored adding this as a pre-submit hook like datcom-ci?
> 
> Should we flush out the plan in the design doc first?

Thanks for review!

This tool now serves as a simplification for current user validation process. The diff capability is not there yet and yes, we should do that in the next step (depending on Hao's progress as he only has 1.5 weeks left for internship).

The screenshot will be triggered on PR review as presubmit results and let author and reviewer verify the screenshot before merge.

**Comment by beets** [comment]
Got it - PR commit and push to dev branch, not merge. Thanks for the clarification!

Some follow ups that will be helpful:
* Listing all the available folders in the UI, so you can easily click to see all the available screenshots
* Show screenshots in two columns: master, and from the dev branch
* Screenshot diffs to catch small alignment issues.

**Comment by shifucun** [comment]
> Got it - PR commit and push to dev branch, not merge. Thanks for the clarification!
> 
> Some follow ups that will be helpful:
> 
> * Listing all the available folders in the UI, so you can easily click to see all the available screenshots
> * Show screenshots in two columns: master, and from the dev branch
> * Screenshot diffs to catch small alignment issues.

Created an issue to track: https://github.com/datacommonsorg/website/issues/281

**Pull request merged**